aptos-move/aptos-vm-types/src/resource_group_adapter.rs

```diff
@@ -176,15 +186,15 @@ impl TResourceGroupView for ResourceGroupAdapter<'_> {
     fn release_group_cache(
         &self,
     ) -> Option<HashMap<Self::GroupKey, BTreeMap<Self::ResourceTag, Bytes>>> {
-        // TODO[agg_v2](fix) - when is_resource_group_split_in_change_set_enabled is false,
+        // TODO[agg_v2](fix) - when is_resource_group_split_in_change_set_capable is false,
         // but self.group_size_kind == GroupSizeKind::AsSum, we need to special-case it.
         // We need to return an indicator here that will force the VM to produce
         // V0 resource group change set but charge as V1
         // Potentially - I'm thinking to turn V0/V1 enum into struct there and prepare both,
         // shouldn't be too complex and well-testable) - we can do that next as one of the
         // preconditions for enabling group feature flag -> aggr2 feature flag.
 
-        if self.is_resource_group_split_in_change_set_enabled()
+        if self.is_resource_group_split_in_change_set_capable()
```

Comment:
this is just default implementation false now, not connected to any feature flags or anything? Shouldn't this be actually based on the provided _enabled flag? (e.g. we could check if the resource group view is set here or smt)

Reply:
this is based on capability (whether resolver/context are capable) and feature flag is whether if capable - it should be enabled.

basically we set to AsSum only if both are true